yktracec/src/jitmodbuilder.cc

```diff
@@ -654,6 +654,7 @@ class JITModBuilder {
     // so that we can insert an appropriate guard into the trace. A block must
     // exist at `InpTrace[TraceIdx + 1]` because the branch instruction must
     // transfer to a successor block, and branching cannot turn off tracing.
+    assert(TraceIdx + 1 < InpTrace.Length());
```

Comment:
The comment explains that is safe, but we may as well keep the assertion in. Agree?

Reply:
Personally I think the `assert` complements the comment, but up to you!

Reply:
Keep it in.